lib/MusicBrainz/Server/Edit/Release/ReorderMediums.pm

```diff
@@ -103,10 +104,28 @@ sub build_display_data {
 
 sub accept {
     my ($self) = @_;
-    $self->c->model('Medium')->reorder(
-        map {
-            $_->{medium_id} => $_->{new}
-        } @{ $self->data->{medium_positions} });
+
+    my %medium_positions = map { $_->{medium_id} => $_->{new} }
+        @{ $self->data->{medium_positions} };
+    my $new_positions = [values %medium_positions];
+
+    my $possible_conflicts =
+        $self->c->sql->select_list_of_hashes(<<'EOSQL', $self->release_id, $new_positions);
+            SELECT id, position
+              FROM medium
+             WHERE release = ?
+               AND position = any(?)
+EOSQL
+
+    for my $row (@$possible_conflicts) {
+        unless (exists $medium_positions{$row->{id}}) {
+            MusicBrainz::Server::Edit::Exceptions::FailedDependency->throw(
+                'Can’t move a medium into position ' . $row->{position} . ', where one already exists.'
+            );
```

Comment:
What happens after this exception is thrown (or in general when edit fails)? I assume ModBot needs to get a note about it somehow.